src/core/src/mapping.rs

```diff
@@ -40,7 +41,14 @@ impl StdError for Error {
 /// Mapping reader
 pub struct Reader<'a, B: Backend, T: 'a + Copy> {
     pub(crate) slice: &'a [T],
-    pub(crate) _mapping: B::Mapping,
+    pub(crate) buffer: &'a B::Buffer,
+    pub(crate) released: bool,
+}
+
+impl<'a, B: Backend, T: 'a + Copy> Drop for Reader<'a, B, T> {
+    fn drop(&mut self) {
+        if !self.released { panic!("a mapping reader was not released"); }
```

Comment:
wouldn't it be more straightforward to just use `assert!(self.released, "a mapping reader was not released")`?

Reply:
I always forget that `assert!` can take messages :)